LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -330,55 +313,61 @@ public void AssignInstructorsToClassrooms()
         public void RegisterStudentsForClasses()
         {
             string action = "Register Students For Classes";
-            int classroomItemListCount = 0;
             ReportHeader(actionName: action);
 
             if (ClassroomList.Count == 0)
             {
-                ReportError(missingResource: "Instructor", failedAction: action);
+                ReportError(missingResource: "Classroom", failedAction: action);
             }
             else
             {
                 foreach (var classroom in ClassroomList)
                 {
-                    if(classroom.ItemList.Count < classroom.MaximumCapacity)
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
                     {
                         UpdateStudentList(classroom);
-                        classroomItemListCount = classroom.ItemList.Count;
-                    } 
-                   
+                    }
+
                 }
             }
 
-            ReportSummary(actionName: action, itemCount: classroomItemListCount);
+            ReportSummary(actionName: action, itemCount: ClassroomList.Count);
             ReportFooter(actionName: action);
         }
 
-        public void Clear()
-        {
-            // Clear all internal data structures 
-            throw new NotImplementedException();
-        }
-
         void UpdateStudentList(Classroom classroom)
         {
             int itemListCount = AttributeGenerator.random.Next(0, classroom.MaximumCapacity - classroom.ItemList.Count + 1);
             int i = 0;
             while (i < itemListCount)
             {
-                foreach (var student in ActiveStudentList)
+                Student studentActive = ActiveStudentList[AttributeGenerator.random.Next(0, ActiveStudentList.Count)];
+
+                if (!studentActive.CompletedModuleList.Contains(classroom.Module) && !studentActive.ClassroomList.Contains(classroom))
                 {
-                    if (!student.CompletedModuleList.Contains(classroom.Module))
-                    {
-                        classroom.ItemList.Add(student);
-                        student.ClassroomList.Add(classroom);
+                    classroom.ItemList.Add(studentActive);
+                    studentActive.ClassroomList.Add(classroom);
 
-                        ReportItem(itemName: student.ToString(), actionName: "Registered for class", itemIndex: i);
-                    }
+                    ReportItem(itemName: studentActive.ToString(), actionName: "Registered for class", itemIndex: i);
                 }
                 i++;
```

Comment:
move i++ inside the **if** block. You should increase the counter only if the student is registered.